fastlane/swift/SocketClient.swift

```diff
@@ -279,24 +309,35 @@ extension SocketClient: StreamDelegate {
         verbose(message: "response is: \(responseString)")
         switch socketResponse.responseType {
         case .clientInitiatedCancel:
-            self.socketDelegate?.commandExecuted(serverResponse: .clientInitiatedCancelAcknowledged)
-            self.closeSession(sendAbort: false)
+            self.socketDelegate?.commandExecuted(serverResponse: .clientInitiatedCancelAcknowledged) {
+                self.closeSession(sendAbort: false)
+                print("Leave \(#function) \(#line)")
+                $0.writeSemaphore.signal()
+            }
+            
 
         case .failure(let failureInformation):
-            self.socketDelegate?.commandExecuted(serverResponse: .serverError)
-            self.handleFailure(message: failureInformation)
+            self.socketDelegate?.commandExecuted(serverResponse: .serverError) {
+                self.handleFailure(message: failureInformation)
+                print("Leave \(#function) \(#line)")
+                $0.writeSemaphore.signal()
+            }
+            
 
         case .parseFailure(let failureInformation):
-            self.socketDelegate?.commandExecuted(serverResponse: .malformedResponse)
-            self.handleFailure(message: failureInformation)
+            self.socketDelegate?.commandExecuted(serverResponse: .malformedResponse) {
+                self.handleFailure(message: failureInformation)
+                print("Leave \(#function) \(#line)")
```

Comment:
Either delete or 👇 ?

```suggestion
                verbose(message: "Leave \(#function) \(#line)")
```